samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplication.java

```diff
@@ -0,0 +1,61 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.runner;
+
+import java.util.List;
+
+import org.apache.samza.SamzaException;
+import org.apache.samza.application.StreamApplication;
+import org.apache.samza.config.Config;
+import org.apache.samza.operators.StreamGraph;
+import org.apache.samza.sql.translator.QueryTranslator;
+import org.apache.samza.sql.testutil.SamzaSqlQueryParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Entry point for the SamzaSQl stream application that takes in SQL as input and performs stream processing.
+ */
+public class SamzaSqlApplication implements StreamApplication {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SamzaSqlApplication.class);
+
+  @Override
+  public void init(StreamGraph streamGraph, Config config) {
+    try {
+      SamzaSqlApplicationConfig sqlConfig = new SamzaSqlApplicationConfig(config);
+      QueryTranslator queryTranslator = new QueryTranslator(sqlConfig);
+      try {
+        Class.forName("org.apache.calcite.jdbc.Driver");
+      } catch (ClassNotFoundException e) {
+        throw new SamzaException("Unable to load calcite jdbc driver");
+      }
+      List<SamzaSqlQueryParser.QueryInfo> queries = sqlConfig.getQueryInfo();
+      for (SamzaSqlQueryParser.QueryInfo query : queries) {
+        LOG.info("Translating the query {} to samza stream graph", query.getSelectQuery());
+        queryTranslator.translate(query, streamGraph);
+      }
+    } catch (Exception e) {
+      LOG.error("SamzaSqlApplication threw exception.", e);
+      throw e;
```

Comment:
How are you able to throw Exception without declaring it?

Reply:
Interesting!!! I didn't notice this, Since this statement just catches and rethrows the exception. Looks like the compiler knows that this routine can only throw UnChecked Exceptions and hence doesn't require declaration. I changed the code to actually catch, log and rethrow just the RuntimeException which was the original intent :).

Reply:
Indeed. I just confirmed your hypothesis. This feature slipped under my radar (clearly I'm not coding in Java enough these days :)).

Reply:
This came in Java 7 apparently: http://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html.